### PR TITLE
Replace certain dependencies (reqwest -> curl + pbr, rulinalg -> ndarray)

### PR DESCRIPTION
**Pull request by quietlychris**
As mentioned by @bytesnake in #7, the `reqwest` crate is fairly heavy in its dependency requirements. To get the same functionality, I've replaced with a combination of curl and [pbr](https://crates.io/crates/pbr) (an ascii progress bar crate) for visualizing download progress.

Furthermore, since the standard `BASE_URL` used by the crate at http://yann.lecun.com/exdb/mnist/ has often ended up being down, I've added a new function to the Builder struct that allows passing a custom base url. I've tested this on cmoran.xyz (with the associated .gz filenames on the end), which seems to work fine. As a result, the datasets that have been traditionally included in this repository have been removed, and any data downloaded to `data/*` should be ignored by git. 

Finally, since rulinalg is no longer maintained and the scientific computing community has seemed to center more around `ndarray`, I've replaced that dependency along with an example of how to convert the MNIST data into an `Array3` data structure, like one might use for testing a neural network. 

Happy to answer any questions or discuss changes which might make sense.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -16,16 +16,18 @@ edition = "2018"
...
-reqwest = {version = "0.10", optional = true, features = ["blocking"]}
+curl = {version = "0.4", optional = true}
+pbr = {version = "1.0", optional = true}
 flate2 = {version = "1.0.2", optional = true, features = ["rust_backend"], default-features = false}
 
 [dev-dependencies]
-rulinalg = "0.4.1"
+ndarray = "0.14"
+image = "0.23"
 # minifb is used to visualize datasets in a pop-up window
```
Comment (bytesnake) [inline Cargo.toml]: update comment

File: examples/fashion_mnist.rs [hunk 2]
```diff
@@ -1,11 +1,8 @@
-extern crate minifb;
-use minifb::{Key, ScaleMode, Window, WindowOptions};
-
 extern crate mnist;
```
Comment (bytesnake) [inline examples/fashion_mnist.rs]: `extern crate` is not required in 2018 rust

File: examples/mnist.rs [hunk 3]
```diff
@@ -0,0 +1,85 @@
+extern crate mnist;
+use image::*;
+use mnist::*;
+use ndarray::prelude::*;
+use show_image::{make_window_full, Event, WindowOptions};
+
+fn main() {
+    let (trn_size, _rows, _cols) = (50_000, 28, 28);
```
Comment (bytesnake) [inline examples/mnist.rs:7]: don't need this as well

File: src/download.rs [hunk 4]
```diff
@@ -36,40 +49,70 @@ pub(super) fn download_and_extract(base_path: &str, use_fashion_data: bool) -> R
...
+                    use std::os::unix::fs::MetadataExt;
+                    let full_size = ARCHIVE_DOWNLOAD_SIZES[i];
+    
+                    let pb_thread = thread::spawn(move || {
+                        let mut pb = ProgressBar::new(full_size.try_into().unwrap());
+                        pb.format("╢=> ╟");
+    
+                        let mut current_size = 0;
+                        while current_size < full_size {
+                            let meta = fs::metadata(file_name.clone())
```
Comment (bytesnake) [inline src/download.rs:97]: nice :+1:

**Review by bytesnake — APPROVED** [review]
cheers :+1:

File: src/download.rs [hunk 5]
```diff
@@ -1,26 +1,39 @@
...
-
 use std::io::{Read, Write};
 use std::path::{Path, PathBuf};
 use std::{fs, io};
 
-const BASE_URL: &str = "http://yann.lecun.com/exdb/mnist";
-const FASHION_BASE_URL: &str = "http://fashion-mnist.s3-website.eu-central-1.amazonaws.com";
+use curl::easy::Easy;
+use std::fs::File;
+// use std::io::{stdout, Write};
```
Comment (davidMcneil) [inline src/download.rs]: Extra line

**Review by davidMcneil — COMMENTED** [review]
Looks good! If you could address the comments; I will merge.

**Comment by quietlychris** [comment]
@davidMcneil  Thank you! The most recent commit should address the comments that you and Lorenz made, but if there's anything else, just let me know!

**Comment by bytesnake** [comment]
ping

**Pull request merged**